spec.md

```diff
@@ -522,6 +544,215 @@ the module.  The `c`{.firrtl} sub-field contained in the `b`{.firrtl} sub-field
 flows into the module, and the `d`{.firrtl} sub-field contained in the
 `b`{.firrtl} sub-field flows out of the module.
 
+
+## Reference Types
+
+References can be exported from a module for indirect access elsewhere, and are
+captured using values of reference type.
+
+For use in cross-module references (hierarchical references in Verilog), a
+reference to a probe of the circuit component is used.  See [@sec:probes] for
+details.
+
+Using reference-type ports, modules may expose internals for reading and
+forcing without routing wires out of the design.
+
+This is often useful for testing and verification, where reference types allow
+reads of the entities to be explicitly exported without hard-coding their place
+in the design.  Instead, by using references, a testbench module may express
+accesses to the internals which will resolve to the appropriate target language
+construct by the compiler (e.g., hierarchical reference).
+
+Reference ports are not expected to be synthesizable or representable in the
+target language and are omitted in the compiled design; they only exist at the
+FIRRTL level.
```

Comment:
Do you mean that they cannot be represented _directly_ in the target language?
I assume the idea is to lower them to something that can be represented, correct?
To me the sentence currently suggests that they are just ignored, which would defeat their purpose.